src/server/logs/logs.go

```diff
@@ -103,12 +109,63 @@ func (ls LogService) GetLogs(ctx context.Context, request *logs.GetLogsRequest,
 						Log: &logs.LogMessage{
 							LogType: &logs.LogMessage_Verbatim{
 								Verbatim: &logs.VerbatimLogMessage{
-									Line: []byte(e.Line),
+									Line:      []byte(e.Line),
+									Timestamp: timestamppb.New(e.Timestamp),
+								},
+							},
+						},
+					},
+				}
+			case logs.LogFormat_LOG_FORMAT_PARSED_JSON:
+				resp = &logs.GetLogsResponse{
+					ResponseType: &logs.GetLogsResponse_Log{
+						Log: &logs.LogMessage{
+							LogType: &logs.LogMessage_Json{
+								Json: &logs.ParsedJSONLogMessage{
+									Verbatim: &logs.VerbatimLogMessage{
+										Line:      []byte(e.Line),
+										Timestamp: timestamppb.New(e.Timestamp),
+									},
+									NativeTimestamp: timestamppb.New(e.Timestamp),
 								},
 							},
 						},
 					},
 				}
+				jsonStruct := new(structpb.Struct)
+				if err := jsonStruct.UnmarshalJSON([]byte(e.Line)); err != nil {
+					log.Error(ctx, "failed to unmarshal json into protobuf Struct", zap.Error(err), zap.String("line", e.Line))
```

Comment:
We should figure out what to do here; I think it would be better to synthesize something like `{"line": e.Line, "timestamp": e.Timestamp}`.  Clients won't be looking at the PPS message response if they requested JSON.